### PR TITLE
[ST] Refactor UserST - add clients to verify messaging

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Refactoring

### Description

This PR refactors the UserST class to improve testing functionality. Changes include the addition of new clients for testing messaging and Kafka user authentication and authorization verification. Also includes refined checks for Kafka user deploy and removed tests that were added as Unit or Integration tests and are no longer required as STs.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=UserST

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=UserST

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java [hunk 1]
```diff
@@ -241,4 +246,47 @@ public static void waitForSpecificLabelKeyValue(String secretName, String namesp
...
+                KafkaResources.clientsCaCertificateSecretName(clusterName)).getData().get("ca.crt");
+            String caKey = KubeClusterResource.kubeClient(namespaceName).getSecret(KafkaResources.clientsCaKeySecretName(clusterName)).getData().get("ca.key");
+
+            File clientCert = OpenSsl.generateSignedCert(csr,
+                                                         SystemTestCertManager.exportCaDataToFile(new String(Base64.getDecoder().decode(caCrt), StandardCharsets.UTF_8), "ca", ".crt"),
+                                                         SystemTestCertManager.exportCaDataToFile(new String(Base64.getDecoder().decode(caKey), StandardCharsets.UTF_8), "ca", ".key"));
+
+            Secret secretBuilder = new SecretBuilder()
+                .withApiVersion("v1")
+                .withKind("Secret")
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java:275]: Can we somehow use constants and void using magic constants like this?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 2]
```diff
@@ -102,57 +92,71 @@ void testUserWithNameMoreThan64Chars(ExtensionContext extensionContext) {
 
         KafkaUserUtils.waitUntilKafkaUserStatusConditionIsPresent(Environment.TEST_SUITE_NAMESPACE, userWithLongName);
 
-        condition = KafkaUserResource.kafkaUserClient().inNamespace(Environment.TEST_SUITE_NAMESPACE).withName(userWithLongName).get().getStatus().getConditions().get(0);
+        Condition condition = KafkaUserResource.kafkaUserClient().inNamespace(Environment.TEST_SUITE_NAMESPACE).withName(userWithLongName).get().getStatus().getConditions().get(0);
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:95]: final?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 3]
```diff
@@ -102,57 +92,71 @@ void testUserWithNameMoreThan64Chars(ExtensionContext extensionContext) {
...
+        KafkaUser kUser = KafkaUserResource.kafkaUserClient().inNamespace(Environment.TEST_SUITE_NAMESPACE).withName(testStorage.getKafkaUsername()).get();
         String kafkaUserAsJson = TestUtils.toJsonString(kUser);
 
-        assertThat(kafkaUserAsJson, hasJsonPath("$.metadata.name", equalTo(userName)));
+        assertThat(kafkaUserAsJson, hasJsonPath("$.metadata.name", equalTo(testStorage.getKafkaUsername())));
         assertThat(kafkaUserAsJson, hasJsonPath("$.metadata.namespace", equalTo(Environment.TEST_SUITE_NAMESPACE)));
         assertThat(kafkaUserAsJson, hasJsonPath("$.spec.authentication.type", equalTo(Constants.TLS_LISTENER_DEFAULT_NAME)));
 
-        long observedGeneration = KafkaUserResource.kafkaUserClient().inNamespace(Environment.TEST_SUITE_NAMESPACE).withName(userName).get().getStatus().getObservedGeneration();
+        long observedGeneration = KafkaUserResource.kafkaUserClient().inNamespace(Environment.TEST_SUITE_NAMESPACE).withName(testStorage.getKafkaUsername()).get().getStatus().getObservedGeneration();
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:121]: final?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 4]
```diff
@@ -171,48 +175,20 @@ void testTlsExternalUserWithQuotas(ExtensionContext extensionContext) {
...
     @ParallelTest
     void testScramUserWithQuotas(ExtensionContext extensionContext) {
         KafkaUser user = KafkaUserTemplates.scramShaUser(Environment.TEST_SUITE_NAMESPACE, userClusterName, "scramed-arnost").build();
-
         testUserWithQuotas(extensionContext, user);
     }
 
-    @ParallelTest
-    void testUserTemplate(ExtensionContext extensionContext) {
+    synchronized void testUserWithQuotas(ExtensionContext extensionContext, KafkaUser user) {
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:189]: Why does this need to be synchronized? I know that's not the aim of this PR but I'm just curious. I glance at this code inside the method I do not see anything, which is not thread-safe. 

1. storageMap is ConcurrentHashMap so this easily handle concurrent access
2. resourceManager is thread-safe by his implementation

and other usages of objects like KafkaCmdClient, KafkaClients, KafkaUserResource, SecretUtils, ClientUtils, and TestUtils do not modify any shared state, which could cause some concurrent problems.

Did you try to execute these tests in parallel? Again such change should be done in another PR but I just wanted to point on it :)

**Review by see-quick — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java:275]: We have constants for that IIRC

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 4 above)
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:189]: This is just diff where i  deleted test, i did not add the synchronized, but i can remove it though, tried it without it, seems to work. Thanks

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java (hunk 1 above)
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java:275]: it is not needed to be specified, as the secret gets created with the builder even without the specified api and kind

File: systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java [hunk 5]
```diff
@@ -191,7 +191,7 @@ public static void waitForCertIsInValidDateRange(File certificate) {
         TestUtils.waitFor("certificate to be in valid date range", Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.CO_OPERATION_TIMEOUT_SHORT,
                           () -> {
                 ZonedDateTime now = ZonedDateTime.now(gmtZone);
-                return (now.isAfter(notBefore.plusSeconds(4)) && now.isBefore(notAfter.minusSeconds(3)));
+                return (now.isAfter(notBefore.plusSeconds(10)) && now.isBefore(notAfter.minusSeconds(10)));
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java:194]: Creating constant with proper name would be good than magic number

**Review by im-konge — APPROVED** [review]

**Pull request merged**